Create CI Workflow File

**Description:** Set up a GitHub Actions (or similar) workflow to run tests on pull requests and commits.  
**Steps:**  
- Create a YAML file under `.github/workflows/`  
- Configure steps to install dependencies and run tests  
- Commit and push the workflow file  
**Estimated Time:** 2 hours

![Image](https://github.com/user-attachments/assets/abdc165f-94eb-459b-b5a1-2a671ae99aa2)

Updation of test results to TEST_RESULTS.md file - Failed
```
Run git config --global user.name "github-actions"
[develop 0f69c75] Update test results in markdown
 1 file changed, 77 insertions(+)
 create mode 100644 TEST_RESULTS.md
To https://github.com/Dhanrajshyam/LittleLemon.git
 ! [rejected]        HEAD -> main (fetch first)
error: failed to push some refs to 'https://github.com/Dhanrajshyam/LittleLemon.git'
hint: Updates were rejected because the remote contains work that you do not
hint: have locally. This is usually caused by another repository pushing to
hint: the same ref. If you want to integrate the remote changes, use
hint: 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -72,12 +72,21 @@ jobs:
         run: |
           git config --global user.name "github-actions"
           git config --global user.email "github-actions@github.com"
+
+          # Extract branch name dynamically
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            TARGET_BRANCH="${{ github.base_ref }}"
+          else
+            TARGET_BRANCH="${GITHUB_REF#refs/heads/}"
+          fi
+
+            echo "Target branch is: $TARGET_BRANCH"
           git add TEST_RESULTS.md
           if git diff --cached --quiet; then
             echo "No changes to commit"
           else
             git commit -m "Update test results in markdown"
-            git push https://${{ secrets.GH_TOKEN }}@github.com/Dhanrajshyam/LittleLemon.git HEAD
+            git push https://${{ secrets.GH_TOKEN }}@github.com/Dhanrajshyam/LittleLemon.git HEAD:$TARGET_BRANCH
           fi
 
       - name: Update README Markdown file
@@ -90,10 +99,18 @@ jobs:
         run: |
           git config --global user.name "github-actions"
           git config --global user.email "github-actions@github.com"
+
+          # Extract branch name dynamically
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            TARGET_BRANCH="${{ github.base_ref }}"
+          else
+            TARGET_BRANCH="${GITHUB_REF#refs/heads/}"
+          fi
+
           git add README.md
           if git diff --cached --quiet; then
             echo "No changes to commit"
           else
             git commit -m "Update readme in markdown"
-            git push https://${{ secrets.GH_TOKEN }}@github.com/Dhanrajshyam/LittleLemon.git HEAD
+            git push https://${{ secrets.GH_TOKEN }}@github.com/Dhanrajshyam/LittleLemon.git HEAD:$TARGET_BRANCH
           fi

